@here/harp-mapview/lib/Utils.ts

```diff
@@ -446,6 +460,229 @@ export namespace MapViewUtils {
         return bytes;
     }
 
+    /**
+     * Computes estimate for size of a THREE.Object3D object and its children. Shared materials
+     * and/or attributes will be counted multiple times.
+     *
+     * @param object The mesh object to evaluate
+     * @param size The [[MemoryUsage]] to update.
+     *
+     * @returns Estimate of object size in bytes for heap and GPU.
+     */
+    export function estimateObject3dSize(
+        object: THREE.Object3D,
+        parentSize?: MemoryUsage
+    ): MemoryUsage {
+        const size =
+            parentSize !== undefined
+                ? parentSize
+                : {
+                      heapSize: 0,
+                      gpuSize: 0
+                  };
+
+        estimateMeshSize(object, size);
+
+        if (object.children.length > 0) {
+            for (const child of object.children) {
+                estimateObject3dSize(child, size);
+            }
+        }
+        return size;
+    }
+
+    function estimateTextureSize(
+        texture: THREE.Texture | null,
+        isVisible: boolean,
+        objectSize: MemoryUsage
+    ) {
+        if (texture === null || texture.image === undefined) {
+            return;
+        }
+
+        // May be HTMLImage or ImageData
+        const image = texture.image;
+        // Assuming RGBA
+        objectSize.heapSize += 4 * image.width * image.height;
+        if (isVisible) {
```

Comment:
Again, even if it's not visible, the texture resource is resident on the GPU.

Reply:
Removed the "isVisible" test, because the handling of shared resources (like THREE.Materials and THREE.Attributes) does not support handling the visibility state of the resource.